src/cockpit/superuser.py

```diff
@@ -99,9 +99,11 @@ class AuthorizeResponder(CockpitResponder):
     def __init__(self, router: Router):
         self.router = router
 
-    async def do_askpass(self, messages: str, prompt: str, hint: str) -> str:
+    async def do_askpass(self, messages: str, prompt: str, hint: str) -> 'str | None':
         hexuser = ''.join(f'{c:02x}' for c in getpass.getuser().encode('ascii'))
-        return await self.router.request_authorization(f'plain1:{hexuser}')
+        password = await self.router.request_authorization(f'plain1:{hexuser}')
+        # translate "no password" from authorize protocol (empty string) to ferny protocol (None)
+        return None if password == '' else password
```

Comment:
What will `sudo` do if we try to sudo as/to a user who actually has an empty password set?

Reply:
You can't actually set an empty password with `passwd`. It does seem to work with `echo 'admin:' | chpasswd` though. You also can't log into ssh with an empty password. `sudo` OTOH doesn't even ask:
```
$ sudo -K
$ sudo whoami
root
```

Same with clicking on the priv indicator: You immediately become root without asking.

So that fringe case seems fine.